lucene/core/src/java/org/apache/lucene/codecs/lucene87/BugfixDeflater_JDK8252739.java

```diff
@@ -40,7 +42,9 @@ public static Deflater createDeflaterInstance(int level, boolean nowrap, int dic
       throw new IllegalArgumentException("dictLength must be >= 0");
     }
     if (IS_BUGGY_JDK) {
-      return new BugfixDeflater_JDK8252739(level, nowrap, dictLength);
+      // Some JDKs need "accessDeclaredMembers" privileges to create a Deflater object
+      return AccessController.doPrivileged(
+              (PrivilegedAction<BugfixDeflater_JDK8252739>) () -> new BugfixDeflater_JDK8252739(level, nowrap, dictLength));

```

Comment:
At least this should be `PrivilegedAction<Deflater>`, because that's what we return.